From ab1c51c2996e5b60152c8b586999d560c53b2f61 Mon Sep 17 00:00:00 2001
From: Huiquan Zhong <huiquan.zhong@intel.com>
Date: Mon, 15 Jun 2015 19:20:27 +0800
Subject: [PATCH] spi/pxa2xx: fix SPI read/write issue after back from S3

Sometimes, while resume back from S3, SPI can't read/write register.
Need to reset the SPI controller first.

Change-Id: I908318502fe92b9d59efdde4a1d040c732cbe688
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-39334
Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
---
 drivers/spi/spi-pxa2xx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-pxa2xx.c b/drivers/spi/spi-pxa2xx.c
index 5482c8ed61f8..07424a86b257 100644
--- a/drivers/spi/spi-pxa2xx.c
+++ b/drivers/spi/spi-pxa2xx.c
@@ -1378,8 +1378,8 @@ static int pxa2xx_spi_resume(struct device *dev)
 	if (!pm_runtime_suspended(dev))
 		clk_prepare_enable(ssp->clk);
 
-	/* Restore LPSS private register bits */
-	lpss_ssp_setup(drv_data);
+	/* Init LPSS private register bits */
+	lpss_ssp_init(drv_data);
 
 	/* Start the queue running */
 	status = spi_master_resume(drv_data->master);
-- 
1.9.1

